plugins/modules/sonic_qos_buffer.py

```diff
@@ -0,0 +1,233 @@
+    type: str
+    choices:
+      - merged
+      - deleted
+    default: merged
+"""
+EXAMPLES = """
+# Using Merged
+#
+# Before state:
+# -------------
+#
+# sonic# show running-configuration | grep buffer
+# buffer init lossless
+# buffer pool ingress_lossless_pool shared-headroom-size 1000000
+
+  - name: Merge QoS buffer configuration
+    dellemc.enterprise_sonic.sonic_qos_buffer:
+      config:
+        buffer_pools:
+          - name: ingress_lossless_pool
+            xoff: 3500000
+        buffer_profiles:
+          - name: profile1
+            pool: ingress_lossless_pool
+            size: 45
+            static_threshold: 25
+            pause_threshold: 55000
+          - name: profile2
+            pool: egress_lossless_pool
+            size: 85
+            dynamic_threshold: -2
+          - name: profile3
+            pool: egress_lossy_pool
+            size: 90
+            static_threshold: 30
+      state: merged
+
+# After state:
+# ------------
+#
+# sonic# show running-configuration | grep buffer
+# buffer init lossless
+# buffer pool ingress_lossless_pool shared-headroom-size 3500000
+# buffer profile profile1 ingress_lossless_pool 45 static-threshold 25 pause pause-threshold 55000
+# buffer profile profile2 egress_lossy_pool 85 dynamic-threshold -2
+# buffer profile profile3 egress_lossless_pool 90 static-threshold 30
+#
+#
+# Using deleted
+#
+# Before state:
+# -------------
+#
+# sonic# show running-configuration | grep buffer
+# buffer init lossless
+# buffer pool ingress_lossless_pool shared-headroom-size 3500000
+# buffer profile profile1 ingress_lossless_pool 45 static-threshold 25 pause pause-threshold 55000
+# buffer profile profile2 egress_lossy_pool 85 dynamic-threshold -2
+# buffer profile profile3 egress_lossless_pool 90 static-threshold 30
+
+  - name: Delete QoS buffer profile configuration
+    dellemc.enterprise_sonic.sonic_qos_buffer:
+      config:
+        buffer_profiles:
+          - name: profile1
+            static_threshold: 25
+            pause_threshold: 55000
+          - name: profile2
+            dynamic_threshold: -2
+          - name: profile3
+      state: deleted
+
+# After state:
+# ------------
+#
+# sonic# show running-configuration | grep buffer
+# buffer init lossless
+# buffer pool ingress_lossless_pool shared-headroom-size 3500000
+# buffer profile profile1 ingress_lossless_pool 45
+# buffer profile profile2 egress_lossy_pool 85
+"""
+RETURN = """
+before:
+  description: The configuration prior to the model invocation.
+  returned: always
+  type: list
+  sample: >
+    The configuration returned will always be in the same format
+    of the parameters above.
+after:
+  description: The resulting configuration model invocation.
+  returned: when changed
+  type: list
+  sample: >
+    The configuration returned will always be in the same format
+    of the parameters above.
```

Comment:
Please add this:
after(generated):
  description: The generated configuration model invocation.
  returned: when C(check_mode)
  type: list
  sample: >
    The configuration returned will always be in the same format
     of the parameters above.